### DISCUSSION TITLE
Can the Pod created using KubernetesExecutor be reused?

### DISCUSSION THREAD

**Discussion by viswanathareddy04** [apache/airflow#16886] [Q&A]
Hi, is there anyway that the pod created in the first execution of task can be reused for the next scheduled run. Instead of creating a new pod for every run. I have a batch job that runs for every minute in this scenario I want the pod to be running and make use of the existing pod. I'm aware its possible in direct K8s, but I want use Airflow for managing my tasks. Please help me out.

**Comment by potiuk** [comment]
You can use CeleryKubernetesExecutor instead. This allows you to specify whether to use K8S executor (which creates separate Pod for each task) or whether to run it through Celery Executor, which can have x workers with y processes that will keep running and will handle tasks from the queue. Shorter tasks in this case could be run via CeleryExecutor (less overhead and more efficient memory/cpu usage) while bigger tasks  where overhead is negligible and separate resource allocation is not a problem as they need it for longer time. 

https://airflow.apache.org/docs/apache-airflow/stable/executor/celery_kubernetes.html
